[Build]  Compile error: "union fp32_bits" defined twice

### Describe the issue

The struct `union fp32_bits` is defined in two places:
* onnxruntime/core/mlas/lib/cast.cpp
* onnxruntime/core/mlas/inc/mlas_float16.h

When I comment out the second definition in `cast.cpp`, the compilation proceeds fine. I don't think cast.cpp needs this second definition of `union fp32_bits` - it's not even used in the file.

### Urgency

_No response_

### Target platform

WebAssembly

### Build script

python ./vendor/onnxruntime/tools/ci_build/build.py \
--build_dir ./build_wasm \
--config=MinSizeRel \
--build_wasm_static_lib \
--parallel \
--minimal_build \
--disable_ml_ops --disable_exceptions --disable_rtti \
--include_ops_by_config ./ort-model/model.required_operators_and_types.config \
--enable_reduced_operator_type_support \
--skip_tests \
--enable_wasm_simd

### Error / output

[ 21%] Linking CXX static library libabsl_stacktrace.a
/home/sevagh/repos/basicpitch.cpp/vendor/onnxruntime/onnxruntime/core/mlas/lib/cast.cpp:18:7: error: redefinition of 'fp32_bits'
   18 | union fp32_bits {
      |       ^
/home/sevagh/repos/basicpitch.cpp/vendor/onnxruntime/onnxruntime/core/mlas/inc/mlas_float16.h:28:7: note: previous definition is here
   28 | union fp32_bits {
      |       ^
[ 21%] Built target absl_throw_delegate
1 error generated.

### Visual Studio Version

_No response_

### GCC / Compiler Version

11.4.0

--- a/onnxruntime/core/mlas/lib/cast.cpp
+++ b/onnxruntime/core/mlas/lib/cast.cpp
@@ -15,11 +15,6 @@ Abstract:
 --*/
 #include "mlasi.h"
 
-union fp32_bits {
-    uint32_t u;
-    float f;
-};
-
 void
 MLASCALL
 MlasConvertHalfToFloatBuffer(

